google-cloud-bigquery/lib/google/cloud/bigquery/project.rb

```diff
@@ -95,6 +95,180 @@ def service_account_email
+        #   * `never` - The table must already exist. A 'notFound' error is
+        #     raised if the table does not exist.
+        # @param [String] write Specifies how to handle data already present in
+        #   the destination table. The default value is `empty`.
+        #
+        #   The following values are supported:
+        #
+        #   * `truncate` - BigQuery overwrites the table data.
+        #   * `append` - BigQuery appends the data to the table.
+        #   * `empty` - An error will be returned if the destination table
+        #     already contains data.
+        # @param [String] job_id A user-defined ID for the copy job. The ID
+        #   must contain only letters (a-z, A-Z), numbers (0-9), underscores
+        #   (_), or dashes (-). The maximum length is 1,024 characters. If
+        #   `job_id` is provided, then `prefix` will not be used.
+        #
+        #   See [Generating a job
+        #   ID](https://cloud.google.com/bigquery/docs/managing-jobs#generate-jobid).
+        # @param [String] prefix A string, usually human-readable, that will be
+        #   prepended to a generated value to produce a unique job ID. For
+        #   example, the prefix `daily_import_job_` can be given to generate a
+        #   job ID such as `daily_import_job_12vEDtMQ0mbp1Mo5Z7mzAFQJZazh`. The
+        #   prefix must contain only letters (a-z, A-Z), numbers (0-9),
+        #   underscores (_), or dashes (-). The maximum length of the entire ID
+        #   is 1,024 characters. If `job_id` is provided, then `prefix` will not
+        #   be used.
+        # @param [Hash] labels A hash of user-provided labels associated with
+        #   the job. You can use these to organize and group your jobs. Label
+        #   keys and values can be no longer than 63 characters, can only
+        #   contain lowercase letters, numeric characters, underscores and
+        #   dashes. International characters are allowed. Label values are
+        #   optional. Label keys must start with a letter and each label in the
+        #   list must have a different key.
+        # @yield [job] a job configuration object
+        # @yieldparam [Google::Cloud::Bigquery::CopyJob::Updater] job a job
+        #   configuration object for setting additional options.
+        #
+        # @return [Google::Cloud::Bigquery::CopyJob]
+        #
+        # @example
+        #   require "google/cloud/bigquery"
+        #
+        #   bigquery = Google::Cloud::Bigquery.new
+        #   dataset = bigquery.dataset "my_dataset"
+        #   destination_table = dataset.table "my_destination_table"
+        #
+        #   bigquery.copy "bigquery-public-data.samples.shakespeare",
+        #                 destination_table
+        #
+        # @!group Data
+        #
+        def copy_job source_table, destination_table, create: nil, write: nil,
+                     dryrun: nil, job_id: nil, prefix: nil, labels: nil
```

Comment:
I recall that setting the dry run property on a non-query job returns an error code. If that's still the case, we should drop this argument.

Reply:
It appears that you are right, when setting `dryrun: true` I see this error: `Google::Cloud::UnimplementedError: Server error`. I will remove the param from this method.

@blowmage Should I deprecate `dryrun` in `Table#copy_job`?

Reply:
I would drop the argument from methods on Project, since these methods have not yet been released. But I would mark the arguments on the methods on Table as deprecated.

Reply:
This is very strange: The `dryrun` param is currently undocumented in both `Table#copy_job` and `Table#extract_job`. I suppose I'll have to add it to the docs just to deprecate it.